modules/cats/src/main/scala/pureconfig/module/package.scala

```diff
@@ -0,0 +1,37 @@
+package pureconfig.module
+
+import scala.language.higherKinds
+import scala.reflect.ClassTag
+
+import _root_.cats.data.{ NonEmptyList, NonEmptyVector }
+import com.typesafe.config.ConfigValue
+import pureconfig.{ ConfigReader, ConfigWriter }
+import pureconfig.ConfigReader.{ fromFunction => fromFunctionReader }
+import pureconfig.ConfigWriter.{ fromFunction => fromFunctionWriter }
+import pureconfig.error.{ ConfigReaderFailures, ConfigValueLocation, EmptyTraversableFound }
+
+/**
+ * [[ConfigReader]] and [[ConfigWriter]] instances for cats data structures.
+ */
+package object cats {
+
+  private[pureconfig] def fromNonEmpty[F[_], G[_], T](fromFT: F[T] => Option[G[T]])(configValue: ConfigValue)(implicit ct: ClassTag[F[T]], fReader: ConfigReader[F[T]]): Either[ConfigReaderFailures, G[T]] =
```

Comment:
Given the very specific signature, which was written to fit `NonEmptyList.fromList` and `NonEmptyVector.fromVector`, I'd leave this here until there was a concrete need for it in the core or by an user. But I'm ok either way :)